visualization/app/codeCharta/ui/toolBar/downloadButton/downloadDialog/util/propertyHelper.ts

```diff
@@ -0,0 +1,37 @@
+import { hierarchy } from "d3-hierarchy"
+import { AttributeTypes, BlacklistItem, BlacklistType, CodeMapNode } from "../../../../../codeCharta.model"
+import { DownloadableSettingsName } from "../../../../../util/fileDownloader"
+
+export type DownloadableProperty = {
+	name: DownloadableSettingsName
+	amount: number
+	isSelected: boolean
+	isDisabled: boolean
+}
+
+export const getDownloadableProperty = (name: DownloadableSettingsName, amount: number): DownloadableProperty => ({
+	name,
+	amount,
+	isSelected: amount > 0,
+	isDisabled: amount === 0
```

Comment:
What happens if amount is negativ? Should we throw an exception or should we play it _fail-safe_ and check `Math.abs(amount)` instead?

Reply:
I don't see why we shouldn't allow to save and upload negative values, if the user had already managed to work with a map with negative values. 

If that would be a problem, I would expect that loading of such a map already throws and therefore downloading it would already be impossible.